pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""This module contains the GraphStatePrep template."""
+
+from typing import Optional, Union
+
+import networkx as nx
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+
+from .qubit_graph import QubitGraph
+
+
+class GraphStatePrep(Operation):
+    r"""
+    Encode a graph state with the specified lattice structure, operations on each qubit, entanglement operations for nearest qubits and qubit graph.
+    The initial graph is :math:`|0\rangle^{\otimes V}`, given each qubit in the graph is in the :math:`|0\rangle` state and is not entangled with each other.
+    The target graph state is:
+    :math:`| \psi \rangle = \prod\limits_{\{a, b\} \in E} U_{ab}|+\rangle^{\otimes V}`
+
+    The target graph state can be prepared as below:
+
+    1. Each qubit is prepared as :math:`|+\rangle^{\otimes V}` state by applying the ``qubit_ops`` (``Hadamard`` gate) operation.
+
+    2. Entangle every nearest qubit pair in the graph with ``entanglement_ops`` (``CZ`` gate) operation.
+
+    Args:
+        graph (Union[QubitGraph, nx.Graph]): QubitGraph or nx.Graph object mapping qubit to wires.
+        qubit_ops (Operation): Operator to prepare the initial state of each qubit. Default to :func:`~.pennylane.H`.
```

Comment:
Sorry I messed up! Gates are classes, not functions :facepalm:. Also don't need to leading `.`

```suggestion
        qubit_ops (Operation): Operator to prepare the initial state of each qubit. Default to :class:`~pennylane.H`.
```